src/platforms/platform.dom.js

```diff
@@ -141,38 +92,93 @@ module.exports = function(Chart) {
 		return canvas;
 	}
 
-	return {
-		/**
-		 * Creates a Chart.js event from a raw event
-		 * @method BrowserPlatform#createEvent
-		 * @implements IPlatform.createEvent
-		 * @param e {Event} the raw event (such as a mouse event)
-		 * @param chart {Chart} the chart to use
-		 * @returns {Core.Event} the chart.js event for this event
-		 */
-		createEvent: function(e, chart) {
-			var relativePosition = helpers.getRelativePosition(e, chart);
-			return {
-				// allow access to the native event
-				native: e,
-
-				// our interal event type
-				type: typeMap[e.type],
-
-				// width and height of chart
-				width: chart.width,
-				height: chart.height,
-
-				// Position relative to the canvas
-				x: relativePosition.x,
-				y: relativePosition.y
-			};
-		},
+	function fromNativeEvent(event, chart) {
+		var position = helpers.getRelativePosition(event, chart);
+		return {
+			native: event,
+			type: eventTypeMap[event.type] || event.type,
+			width: chart.width,
+			height: chart.height,
+			x: position.x,
+			y: position.y
+		};
+	}
+
+	function createResizer(handler) {
+		var iframe = document.createElement('iframe');
+		iframe.className = 'chartjs-hidden-iframe';
+		iframe.style.cssText =
+			'display:block;'+
+			'overflow:hidden;'+
+			'border:0;'+
+			'margin:0;'+
+			'top:0;'+
+			'left:0;'+
+			'bottom:0;'+
+			'right:0;'+
+			'height:100%;'+
+			'width:100%;'+
+			'position:absolute;'+
+			'pointer-events:none;'+
+			'z-index:-1;';
+
+		// Prevent the iframe to gain focus on tab.
+		// https://github.com/chartjs/Chart.js/issues/3090
+		iframe.tabIndex = -1;
+
+		// If the iframe is re-attached to the DOM, the resize listener is removed because the
+		// content is reloaded, so make sure to install the handler after the iframe is loaded.
+		// https://github.com/chartjs/Chart.js/issues/3521
+		helpers.addEvent(iframe, 'load', function() {
+			helpers.addEvent(iframe.contentWindow || iframe, 'resize', handler);
+
+			// The iframe size might have changed while loading, which can also
+			// happen if the size has been changed while detached from the DOM.
+			handler();
+		});
+
+		return iframe;
+	}
+
+	function addResizeListener(node, listener) {
+		var stub = node._chartjs = {
+			ticking: false
+		};
+
+		// Throttle the callback notification until the next animation frame.
+		var notify = function() {
+			if (!stub.ticking) {
+				stub.ticking = true;
+				helpers.requestAnimFrame.call(window, function() {
+					if (stub.resizer) {
+						stub.ticking = false;
+						return listener();
```

Comment:
Just realized that 'resize' listeners are not given an event object. Not  a big deal since we don't use it, but would be more consistent with the other events. What do you think?